build-tools/installers/sign-pkg-content.proj

```diff
@@ -0,0 +1,66 @@
+<!--
+***********************************************************************************************
+sign-pkg-content.proj
+
+This project file is used to sign and harden all relevent files included in both our legacy
+and .NET 6 .pkg installers.  ESRP signing does not yet support passing a runtime entitlements
+file alongside a file that needs runtime hardening, so we have to apply the entitlements
+ourself (using an empty signing identity) before passing these files to ESRP.
+***********************************************************************************************
+-->
+<Project Sdk="Microsoft.Build.NoTargets">
+
+  <Import Project="..\..\Configuration.props" />
+  <Import Project="create-installers.targets" />
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <GenerateDependencyFile>false</GenerateDependencyFile>
+    <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
+    <!-- NOTE: Any file you want to sign must be under either $(OutDir) or $(IntermediateOutputPath). -->
+    <OutDir Condition=" '$(OutDir)' == '' ">$(MSBuildSrcDir)</OutDir>
+    <_BundleToolJar>$([System.IO.Path]::GetFullPath('$(MSBuildSrcDir)\bundletool.jar'))</_BundleToolJar>
+    <_BundleToolExtractLocation>$(MSBuildSrcDir)\sign-bundletool</_BundleToolExtractLocation>
+    <_BundleToolAapt2>$(_BundleToolExtractLocation)\macos\aapt2</_BundleToolAapt2>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.VisualStudioEng.MicroBuild.Core" Version="$(MicroBuildCoreVersion)">
+      <PrivateAssets>all</PrivateAssets>
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers</IncludeAssets>
+    </PackageReference>
+  </ItemGroup>
+
+  <Target Name="_AddMachOEntitlements"
+      BeforeTargets="SignFiles"
+      AfterTargets="AfterBuild" >
+    <Exec Command="codesign -vvvv -f -s - -o runtime --entitlements &quot;%(_MSBuildFilesUnixSignAndHarden.EntitlementsPath)&quot; &quot;%(_MSBuildFilesUnixSignAndHarden.Identity)&quot;" />
+    <!-- Unpack bundletool.jar and sign embedded aapt2 -->
+    <RemoveDir Directories="$(_BundleToolExtractLocation)" />
```

Comment:
Just a thought, I didn't know you were having to deal with `aapt2` inside `bundletool`.

We always pass a path to `aapt2` to `bundletool`, so it uses *our* `aapt2`, I wonder if we could simply remove it from `bundletool.jar`? I guess that would invalidate `bundletoo.jar`'s signature.

Reply:
I like this idea, we can save some build time (and installation space) by stripping out the nested aapt2 rather than signing it. Let's see if this causes any test failures.

Reply:
Looks like bundletool still expects aapt2 to exist even if it may not be using it in our case, I think I'll revert these latest changes:
```
[BT : 1.4.0] error : Unable to locate aapt2 inside jar. [/Users/runner/work/1/s/tests/Mono.Android-Tests/Runtime-AppBundle/Mono.Android-TestsAppBundle.csproj]
  com.android.tools.build.bundletool.model.exceptions.CommandExecutionException: Unable to locate aapt2 inside jar.
  	at com.android.tools.build.bundletool.model.exceptions.InternalExceptionBuilder.build(InternalExceptionBuilder.java:57)
  	at com.android.tools.build.bundletool.model.utils.SdkToolsLocator.extractAapt2(SdkToolsLocator.java:109)
  	at com.android.tools.build.bundletool.commands.CommandUtils.extractAapt2FromJar(CommandUtils.java:64)
  	at com.android.tools.build.bundletool.commands.BundletoolModule.lambda$provideAapt2Command$0(BundletoolModule.java:35)
  	at java.base/java.util.Optional.orElseGet(Optional.java:369)
  	at com.android.tools.build.bundletool.commands.BundletoolModule.provideAapt2Command(BundletoolModule.java:35)
  	at com.android.tools.build.bundletool.commands.BundletoolModule_ProvideAapt2CommandFactory.provideAapt2Command(BundletoolModule_ProvideAapt2CommandFactory.java:40)
  	at com.android.tools.build.bundletool.commands.BundletoolModule_ProvideAapt2CommandFactory.get(BundletoolModule_ProvideAapt2CommandFactory.java:31)
  	at com.android.tools.build.bundletool.commands.BundletoolModule_ProvideAapt2CommandFactory.get(BundletoolModule_ProvideAapt2CommandFactory.java:10)
  	at dagger.internal.DoubleCheck.get(DoubleCheck.java:47)
```

Reply:
Do you know the command-line that hit this? Just want to check we don't have something wrong.

Reply:
This happened in the apk tests job during `DeployTestAabs`: https://devdiv.visualstudio.com/DevDiv/_build/results?buildId=4555981&view=logs&j=cd6eb242-2e57-569b-620a-eaaded8e9a51&t=f5527f3e-27f9-5958-e19f-f922328304e1&l=715. My initial changes to strip aapt2 out only ran when doing real signing so the full set of tests didn't run against it. I can kick something off against that commit to see if anything else breaks.

Reply:
Yeah weird we pass `--adb` and they still extract it:
```
/Users/runner/Library/Android/jdk-11/bin/java -jar /Library/Frameworks/Xamarin.Android.framework/Libraries/xbuild/Xamarin/Android/bundletool.jar build-apks --connected-device --overwrite --mode default --bundle "../../../bin/TestRelease/Mono.Android_TestsAppBundle-Signed.aab" --output "../../../bin/TestRelease/Mono.Android_TestsAppBundle-Signed.aab.apks" --adb /Users/runner/Library/Android/sdk/platform-tools/adb --ks /Users/runner/work/1/s/src/Xamarin.Android.Build.Tasks/Tests/Xamarin.ProjectTools/Resources/Base/test.keystore --ks-key-alias mykey --key-pass pass:android --ks-pass pass:android 
[BT : 1.4.0] error : Unable to locate aapt2 inside jar.
```